### PR TITLE
Fix args parsing - RHMAP-16759

### PR DESCRIPTION
**Pull request by evanshortiss**
Addresses #383

### CODE REVIEW

File: lib/utils/args.js [hunk 1]
```diff
@@ -54,21 +54,24 @@ function _normaliseFlagsAndArs(argsIntoCommand, cmd) {
     // bail on old style commands
     return argsIntoCommand;
   }
-  var processArgs = _.clone(process.argv);
+
+  const normalisedArgs = _.clone(argsIntoCommand);
+
```
Comment (fh-qube) [inline lib/utils/args.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: lib/utils/args.js [hunk 2]
```diff
@@ -54,21 +54,24 @@ function _normaliseFlagsAndArs(argsIntoCommand, cmd) {
...
+        var idx = normalisedArgs.indexOf(arg);
         if (idx > -1) {
-          processArgs[idx] = '--' + alias + '=' + arg;
+          normalisedArgs[idx] = '--' + alias + '=' + arg;
         }
       }
     }
   });
-  return processArgs;
+
```
Comment (fh-qube) [inline lib/utils/args.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: lib/utils/args.js [hunk 3]
```diff
@@ -54,21 +54,24 @@ function _normaliseFlagsAndArs(argsIntoCommand, cmd) {
     // bail on old style commands
     return argsIntoCommand;
   }
-  var processArgs = _.clone(process.argv);
+
+  const normalisedArgs = _.clone(argsIntoCommand);
```
Comment (evanshortiss) [inline lib/utils/args.js]: Need to use the passed arguments here, not `process.argv` since `process.argv` only works if using as  CLI. This new method works for both CLI and as a module.

**Comment by camilamacedo86** [comment]
Hi @evanshortiss,
Tks for the contribution. Could you do the following actions for we are able to move forward with it?

1. Could you rebase the PR with the master?
2. Could you describe the steps to test it?
3. Could you squash the both commits into just one as "JIRA - Text"?
 
After you do this steps I will test and if works I will approve your PR and merge it into the master.

**Comment by evanshortiss** [comment]
Thanks @camilamacedo86 - squash and rebase is done 😄 

Steps for testing are available in issue #383, but I can provide more detail if required.

**Comment by evanshortiss** [comment]
Hmm, actually let me rephrase the steps in #383.

After this PR is merged we should be able to run commands from the CLI like so:

```
fhc projects read --project=$SOME_PROJECT_ID
```

and as a node.js script like so:

```js
const fhc = require('fh-fhc')

fhc.load((err) => {
  if (err) throw err

  fhc.applyCommandFunction(['projects', 'read', '--project=SOME_PROJECT_ID'], console.log.bind(console))
})
```

**Comment by camilamacedo86** [comment]
Hi @evanshortiss,

This PR is working for the example that you wrote here as follows.

` fhc.applyCommandFunction(['projects', 'read', '--project=hgxe4z6k3qxgpolko4ry7fik'], console.log.bind(console));
`

However, it is not working when is required pass more than one parameter as follows.
`
 fhc.applyCommandFunction(['env', 'list', '--app=hgxe4z5j2b6owln5vw6qvx5i --env=dev'], console.log.bind(console));`
 
Result: 

```
{ [Error: Error - not 2xx status code. (400)
{"message":"expected a guid :invalid guid char: [ ]","status":"error","result":null}] requestId: '1a38e534-0a35-4f11-ad34-b981ee8d55ef' }
fhc ERR! cb() never called!
fhc ERR!  
fhc Command executed with error
```

Also, it is not solving the situation where the V3 standard can not receive a sequence array as I understood that was your intention here as follows. 

```
    fhc.env.list({_:['hgxe4z5j2b6owln5vw6qvx5i', 'dev'], json : true }, function (err,res) {
      return cb( err);
    });
```

Result:

```
fhc ERR! Error: Error - not 2xx status code. (400)
fhc ERR! {"message":"expected a guid :Invalid guid length, 9 = 4*2+1, only +2 or +3 allowed","status":"error","result":null}
fhc ERR! Request ID:  8eaa9d93-c741-479a-bde5-7dd97b7fc6b7
fhc ERR! 
fhc ERR! System Darwin 15.6.0
fhc ERR! command "/Users/cmacedo/.nvm/versions/node/v4.3.0/bin/node" "/Users/cmacedo/work/fh-fhc/bin/fhc.js" "test"
```

The tests made was with the command `fhc env list --app=<app> [--env=<env>] [--json=<json>]
`
Following the result of this command into the command line.

dhcp-17-108:fh-fhc cmacedo$ ./bin/fhc.js env list hgxe4z5j2b6owln5vw6qvx5i dev true
[{
  "businessObject": "cluster/reseller/customer/domain/project/cloud-apps/environment-variable",
  "domain": "support",
  "appId": "hgxe4z5j2b6owln5vw6qvx5i",
  "varName": "asdasd",
  "varValues": {
    "dev": "adasd"
  },
  "masked": false,
  "guid": "kfoo52j2hegdtnxip2u7jdyh",
  "sysOwner": "i4ze7ruomd27wnohth22to4v",
  "sysVersion": 0,
  "sysCreated": 1501702889102,
  "sysModified": 1501702889114,
  "sysGroupFlags": 65567,
  "sysGroupList": ""
}]
dhcp-17-108:fh-fhc cmacedo$ 

Following how we are expecting that it be called as a dependency which is working.

```
    fhc.env.list({app:'hgxe4z5j2b6owln5vw6qvx5i', env:'dev', json : true }, function (err,res) {
      return cb( res);
    });
```

File: lib/utils/args.js (hunk 3 above)
Comment (camilamacedo86) [inline lib/utils/args.js]: For the cases where more than one parameter is required, it doesn't work well expected for me. Could you check the tests performed and tell me your position?

**Comment by camilamacedo86** [comment]
@evanshortiss I tried to test as you can check above. Could you validate it? Please, let me know if I did something wrong or if you need some help in order to check it for we move forward.

**Comment by camilamacedo86** [comment]
@evanshortiss could verify the review? Is it something that we should still consider or could we close this one?

**Comment by camilamacedo86** [comment]
@evanshortiss we are looking for your feedback here. Should we keep this PR and issue open or we can close them? If it is to be kept open could you check the review made?

**Comment by camilamacedo86** [comment]
Consing this PR since it is not so long passing in the Jenkins, CI process and shows outdated without any interaction for a long time.
